drivers/flash/flash_stm32.c

```diff
@@ -34,19 +34,33 @@
 #define STM32_FLASH_TIMEOUT	(K_MSEC(25))
 #endif
 
+#define CFG_HW_FLASH_SEMID	2
+
 /*
  * This is named flash_stm32_sem_take instead of flash_stm32_lock (and
  * similarly for flash_stm32_sem_give) to avoid confusion with locking
  * actual flash pages.
  */
 static inline void flash_stm32_sem_take(struct device *dev)
 {
+
+#ifdef CONFIG_SOC_SERIES_STM32WBX
+	while (LL_HSEM_1StepLock(HSEM, CFG_HW_FLASH_SEMID))
```

Comment:
I'm quite sure this could be defined as a standalone hardware semaphore driver and defined in DT instead of hardcoding usage here

Reply:
Agreed. Can we leave this as future work?
If you agree, I'll create an issue to track it.